tls/s2n_x509_validator.h

```diff
@@ -32,6 +32,7 @@
 typedef enum {
     UNINIT,
     INIT,
+    PRE_VALIDATE,
     VALIDATED,
```

Comment:
I don't love this name. How about "PARSED"? Or "CERT_READ"?

Reply:
I chose something like `PRE_VALIDATE`, because this stage kind of does more than just parse the cert chain. It also calls `s2n_verify_host_information`, and will also call `s2n_crl_lookup` in the next PR. How about `CERT_CHAIN_PROCESSED`? I added this in a9c024edb444cfa2aec507ad3ccb85dd16792ea4. Let me know if you don't like this name.

Also, switching to the `PRE_VALIDATE` state sort of implied that `skip_cert_validation` was false, so I checked for this before switching states. With `CERT_CHAIN_PROCESSED`, this isn't clear, So in a9c024edb444cfa2aec507ad3ccb85dd16792ea4 I made it always enter this state and skip the verify parts in `s2n_x509_validator_verify_cert_chain`.

Reply:
Maybe "READY_TO_VALIDATE"? Or just "READY"? Names are hard :\

Reply:
READY_TO_VALIDATE makes sense. I switched it to READY_TO_VERIFY, to keep the state name the same as `s2n_x509_validator_verify_cert_chain`, which does the actual X509_verify_cert call now. 876a33b049c2daa464edd07e58f53901009fe26c